exporter.go

```diff
@@ -169,3 +207,82 @@ func (e *Exporter) addLayer(image imgutil.Image, layer identifiableLayer, previo
 	e.Logger.Debugf("Layer '%s' SHA: %s\n", layer.Identifier(), sha)
 	return sha, image.AddLayer(tarPath)
 }
+
+func (e *Exporter) createSliceLayer(image imgutil.Image, layerID string, files []string) (string, identifiableLayer, error) {
+	tarPath := filepath.Join(e.ArtifactsDir, escapeID(layerID)+".tar")
+	sha, fileSet, err := archive.WriteFilesToTar(tarPath, e.UID, e.GID, files...)
+	if err != nil {
+		return "", nil, errors.Wrapf(err, "exporting slice layer '%s'", layerID)
+	}
+
+	// for this first iteration, just delete the actual files then revisit
+	// the directories and delete if empty as a result of previous removal
+	var dirs []string
+	for file := range fileSet {
+		stat, _ := os.Stat(file)
+		if !stat.IsDir() {
+			err = os.Remove(file)
+			if err != nil {
+				e.Logger.Errorf("failed to delete file %v", err)
+			}
+		} else {
+			dirs = append(dirs, file)
+		}
+	}
+	for _, dir := range dirs {
+		if ok, _ := IsEmptyDir(dir); ok {
+			err = os.Remove(dir)
+			if err != nil {
+				e.Logger.Errorf("failed to delete directory %v", err)
+			}
+		}
+	}
+
+	return sha, &layer{path: tarPath, identifier: layerID}, nil
+}
+
+func (e *Exporter) addSliceLayers(image imgutil.Image, sliceLayers map[string]identifiableLayer, previousLayer []LayerMetadata) ([]LayerMetadata, error) {
+	var addedSliceLayers []LayerMetadata
+	for sha, layer := range sliceLayers {
+		var err error
+
+		found := false
+		for _, previous := range previousLayer {
+			if sha == previous.SHA {
+				found = true
+				break
+			}
+		}
+		if found {
+			e.Logger.Infof("Reusing layer '%s'\n", layer.Identifier())
+			e.Logger.Infof("Layer '%s' SHA: %s\n", layer.Identifier(), sha)
+			err = image.ReuseLayer(sha)
+		} else {
+			e.Logger.Infof("Adding layer '%s'\n", layer.Identifier())
+			e.Logger.Infof("Layer '%s' SHA: %s\n", layer.Identifier(), sha)
```

Comment:
Given the current output:

```
===> EXPORTING
[exporter] Reusing layer 'launcher'
[exporter] Reusing layer 'io.pivotal.openjdk:class-counter'
[exporter] Reusing layer 'io.pivotal.openjdk:java-security-properties'
[exporter] Reusing layer 'io.pivotal.openjdk:jvmkill'
[exporter] Reusing layer 'io.pivotal.openjdk:link-local-dns'
[exporter] Reusing layer 'io.pivotal.openjdk:memory-calculator'
[exporter] Reusing layer 'io.pivotal.openjdk:openjdk-jre'
[exporter] Reusing layer 'io.pivotal.openjdk:security-provider-configurer'
[exporter] Reusing layer 'io.pivotal.jvmapplication:executable-jar'
[exporter] Reusing layer 'io.pivotal.springboot:spring-boot'
[exporter] Reusing layer 'io.pivotal.clientcertificatemapper:client-certificate-mapper'
[exporter] Reusing layer 'io.pivotal.containersecurityprovider:container-security-provider'
[exporter] Reusing layer 'io.pivotal.springautoreconfiguration:auto-reconfiguration'
[exporter] Reusing layer 'slice-3'
[exporter] Layer 'slice-3' SHA: sha256:a90f94780489ffb2a74d973a9bea7a8b0c1c3df0b1703c9408dfc65e2649aaf8
[exporter] Reusing layer 'slice-4'
[exporter] Layer 'slice-4' SHA: sha256:55a0304911d6eb9f4338f2fa9a9cfe9291c83125af151b7e8155b7b10615e8c7
[exporter] Reusing layer 'slice-5'
[exporter] Layer 'slice-5' SHA: sha256:aa0914e173b579980f29a25bc151fc5494cdc0e0c4676109f5340bf7e8f830a3
[exporter] Reusing layer 'slice-1'
[exporter] Layer 'slice-1' SHA: sha256:48e99c4ff85a50bb2ec6bb1feceadbe5628d3227ebce111bf3ed2f5702b2f115
[exporter] Reusing layer 'slice-2'
[exporter] Layer 'slice-2' SHA: sha256:a30d09749a148759df26eaca697dee994db3c37adde6e68406f57fe54e76e573
[exporter] Adding layer 'app'
[exporter] Reusing layer 'config'
[exporter] *** Images (b1b33820c01a):
[exporter]       index.docker.io/applications/jar-source:latest
```

and the [existing code](https://github.com/buildpack/lifecycle/blob/master/exporter.go#L96), this like should be `Debugf`.

Reply:
Looks good in latest commit.